pkg/apis/provisioning/v1alpha5/labels.go

```diff
@@ -34,6 +34,10 @@ var (
 	KarpenterLabelDomain = "karpenter.sh"
 	LabelCapacityType    = KarpenterLabelDomain + "/capacity-type"
 
+	// AnnotationExtendedResources is used to record the expected extended resources on a node that will be created when
```

Comment:
We need to know about extended resources that should eventually appear on the node so we can schedule against it as an in-flight node (e.g. we need to know there will be 4 NVidia GPUs so we don't launch another GPU instance due to a pending GPU pod).  We can't set the resources on the node as then kube-scheduler races with kubelet and sometimes schedules pods before kubelet zeros resources which leads to an eviction.